client/components/scoresheet/metadata_inputs/metadata_inputs_controller.js

```diff
@@ -146,17 +147,22 @@ class MetadataInputsController {
     if (!this.data.autoselect) {
       return Promise.resolve()
     }
-    if (this.data.current.matchId !== undefined) {
-      return Promise.resolve()
-    }
-    if (this.data.current.round !== undefined && this.data.current.stage !== undefined) {
-      return Promise.resolve()
-    }
-    if (this.matches.every(match => this.data.current.stage !== match.stage || this.data.current.round !== match.round)) {
-      const firstIncompleteMatch = this.matches.find(match => !match.complete)
-      if (firstIncompleteMatch) {
-        this.data.current.matchId = firstIncompleteMatch._id
+    if (!this.forceMatchReselection) {
+      if (this.data.current.matchId !== undefined) {
+        return Promise.resolve()
       }
+      if (this.data.current.round !== undefined && this.data.current.stage !== undefined) {
+        return Promise.resolve()
+      }
+      if (this.matches.some(match => this.data.current.stage === match.stage || this.data.current.round === match.round)) {
+        return Promise.resolve()
+      }
+    }
+    this.forceMatchReselection = false
+    const firstIncompleteMatch = this.matches.find(match => !match.complete)
+    if (firstIncompleteMatch) {
+      this.data.current.stage = firstIncompleteMatch.stage
+      this.data.current.round = firstIncompleteMatch.round
```

Comment:
this function doesn't return promise in all of it's paths

Reply:
@2roy999 did you mean to approve the PR? This sounds important to me

Reply:
It's not really important in the way it's used now, but I fixed it anyway

Reply:
It's not really important in the way it's used now, but I fixed it anyway